backend/src/services/auth.rs

```diff
@@ -0,0 +1,349 @@
+use super::prelude::*;
+
+pub type HttpClient = ContextWrapper<
+    Client<
+        DropContextService<hyper::Client<HttpConnector, Body>, ClientContext>,
+        ClientContext,
+        Basic,
+    >,
+    ClientContext,
+>;
+pub type HttpBobClient = BobClient<HttpClient>;
+
+#[derive(Debug, Error)]
+pub enum AuthError {
+    #[error("Couldn't load user")]
+    LoadError,
+    #[error("Couldn't lock user store")]
+    PoisonError,
+    #[error("Auth layer failed to deserialize data")]
+    DeserializeError,
+    #[error("Couldn't login user")]
+    LoginError,
+    #[error("Couldn't logout user")]
+    LogoutError,
+    #[error("Couldn't update session")]
+    SessionError,
+}
+
+/// Optional credentials for a BOB cluster
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Default, Deserialize)]
+pub struct Credentials {
+    pub login: String,
+    pub password: String,
+}
+
+/// Login to a BOB cluster
+///
+/// # Errors
+/// This function can return the following errors
+///
+/// 1. [`StatusCode::BAD_REQUEST`]
+/// The function failed to parse hostname of the request
+///
+/// 2. [`StatusCode::NOT_FOUND`]
+/// The client was unable to reach the host
+///
+/// 3. [`StatusCode::UNAUTHORIZED`]
+/// The client couldn't authorize on the host
+///
+#[cfg_attr(feature = "swagger",
+    utoipa::path(
+        post,
+        context_path = ApiV1::to_path(),
+        path = "/login",
+        params(
+            BobConnectionData
+        ),
+        responses(
+            (status = 200, description = "Successful authorization"),
+            (status = 400, description = "Bad Hostname"),
+            (status = 401, description = "Bad Credentials"),
+            (status = 404, description = "Can't reach specified hostname")
+        )
+))]
+pub async fn login(
+    mut auth: BobAuth<HttpBobClient>,
+    Extension(request_timeout): Extension<RequestTimeout>,
+    Json(bob): Json<BobConnectionData>,
+) -> AxumResult<StatusCode> {
+    tracing::info!("post /login : {:?}", &bob);
```

Comment:
`bob` contains `password` inside. Writing password to logs is a vary bad practice. Please change `Debug` implementation on `Credentials` so that it doesn't write the password